apps/user/views.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework.views import APIView
+from rest_framework.authtoken.views import ObtainAuthToken
+from rest_framework.response import Response
+
+from .models import (
+    ExpiringToken,
+    User,
+    UserProfile,
+)
+from .serializers import (
+    RegistrationSerializer,
+)
+
+
+class ObtainExpiringAuthToken(ObtainAuthToken):
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data['user']
+
+        token, created = ExpiringToken.objects.get_or_create(user=user)
+
+        if not created:
+            if token.expired():
+                token.delete()
+                token = ExpiringToken.objects.create(user=user)
```

Comment:
It will match better to use `generate_key` method of `Token` class to update the current value instead of recreating the whole object. 
Exactly this method uses when generating `key` field value while saving a new token object.